includes/class-platform-checkout-tracker.php

```diff
@@ -154,20 +165,22 @@ public function should_enable_tracking() {
 	/**
 	 * Record an event in Tracks - this is the preferred way to record events from PHP.
 	 *
-	 * @param mixed  $user                   username, user_id, or WP_user object.
 	 * @param string $event_name             The name of the event.
 	 * @param array  $properties             Custom properties to send with the event.
+	 * @param bool   $is_admin_event         Indicate whether the event is emitted from admin area.
 	 *
 	 * @return bool|array|\WP_Error|\Jetpack_Tracks_Event
 	 */
-	public function tracks_record_event( $user, $event_name, $properties = [] ) {
+	public function tracks_record_event( $event_name, $properties = [], $is_admin_event = false ) {
+
+		$user = wp_get_current_user();
 
 		// We don't want to track user events during unit tests/CI runs.
 		if ( $user instanceof \WP_User && 'wptests_capabilities' === $user->cap_key ) {
 			return false;
 		}
 
-		if ( ! $this->should_enable_tracking() ) {
+		if ( ! $is_admin_event && ! $this->should_enable_tracking() ) {
```

Comment:
`is_admin_event` identifies the events coming through the `maybe_record_admin_event`. This check makes sure that those requests bail out early without considering the conditions in `should_enable_tracking` method, as we want these events to be tracked always. 
But after you pointed it out, I realized it would make sense to consolidate all the validations in `should_enable_tracking`, so I moved this check inside that function in https://github.com/Automattic/woocommerce-payments/pull/6175/commits/36da0f3ddfe323c5e54646fd76c90f634d02fdee

Reply:
What I meant is that it sounds strange to have checks in the function with a name that makes it look like it will execute something no matter what and have no checks in the functions with a name that indicate it may or may not execute. It's just a nitpicking and can be ignored.